general/mem_manager.cpp

```diff
@@ -869,37 +869,40 @@ void MemoryManager::SetDeviceMemoryType_(void *h_ptr, unsigned flags,
    }
 }
 
-MemoryType MemoryManager::Delete_(void *h_ptr, MemoryType mt, unsigned flags)
+MemoryType MemoryManager::Delete_(void *h_ptr, MemoryType h_mt, unsigned flags)
 {
    const bool alias = flags & Mem::ALIAS;
    const bool registered = flags & Mem::REGISTERED;
    const bool owns_host = flags & Mem::OWNS_HOST;
    const bool owns_device = flags & Mem::OWNS_DEVICE;
    const bool owns_internal = flags & Mem::OWNS_INTERNAL;
-   MFEM_ASSERT(registered || IsHostMemory(mt),"");
+   MFEM_ASSERT(IsHostMemory(h_mt), "invalid h_mt = " << (int)h_mt);
+   // MFEM_ASSERT(registered || IsHostMemory(h_mt),"");
    MFEM_ASSERT(!owns_device || owns_internal, "invalid Memory state");
-   if (!mm.exists || !registered) { return mt; }
```

Comment:
Are we sure removing this will not break anything? Could we warn or add a VERIFY that could be triggered in DEBUG mode?

Reply:
This is actually not removed, see here: https://github.com/mfem/mfem/blob/b7ab371c805d7f7c88303a31eeafb33c15a164d3/general/mem_manager.cpp#L884
I think github shows it as removed because `mt` was renamed to `h_mt`, among the other additions.